core/psh/sysexec.c

```diff
@@ -25,6 +27,63 @@ void psh_sysexecinfo(void)
 }
 
 
+static int psh_sysexec_wldargmatch(char *wld, int argc, char **argv, char wldend, int argmaxlen)
+{
+	int len = 0, ii;
+	char *currc = wld, *nextc, *cmdend;
+
+	cmdend = strchr(wld, wldend);
+	currc = wld;
+	for (ii = 0; ii < argc; ii++) {
+
+		if ((nextc = strchr(currc, ' ')) == NULL || nextc > cmdend)
+			nextc = cmdend;
+		if ((nextc == cmdend) ^ (ii == (argc - 1))) /* exits if: last argc XOR last wldcrd argument */
+			return -1;
+
+		len = nextc - currc;
+		if (len > argmaxlen || len == 0 || strlen(argv[ii]) != len || memcmp(argv[ii], currc, len) != 0)
+			return -1;
+		if (*nextc != wldend && nextc != NULL)
+			currc = nextc + 1;
+	}
+	
+	return 1;
+}
+
+
+static int psh_sysexec_checkcommand(int argc, char **argv)
+{
+	int maxcdmlen = 80, acc = -1;
+	char wldbuff[maxcdmlen], *wldlist = PSH_WILDCARDS, *curr = wldlist, *nxtcomma;
+	FILE *wldfile;
+
+	/* check against /etc/wldcrds file */
+	if ((wldfile = fopen("/etc/wldcrds", "r")) != NULL) {
+		acc = 0;
+		while (fgets(wldbuff, maxcdmlen, wldfile) != NULL && acc != 1) {
+			if (!isprint(wldbuff[maxcdmlen - 2]) && psh_sysexec_wldargmatch(wldbuff, argc, argv, '\n', maxcdmlen-1) > 0)
+				acc = 1;
+			else
+				while (isprint(fgetc(wldfile))); /* skip to next wildcard */
+				
+			memset(wldbuff, '\0', sizeof(wldbuff));
+		}
+		fclose(wldfile);
+	}
+	
+	/* check PSH_WILDCARDS flag */
+	while ((nxtcomma = strchr(curr, ';')) != NULL && acc != 1) {
+		acc = 0;
+		if (!acc && (psh_sysexec_wldargmatch(curr, argc, argv, ';', (nxtcomma - curr + 1))) > 0)
```

Comment:
what? why checking `acc` if You've set it to 0 in the previous line? Also: why setting to 0?

Reply:
Also: why passing `argmaxlen` as a total length of the wildcard? seems invalid (the other call does the same) - what was the intent?

Reply:
#### About `acc`:
`acc` value at the time of `psh_sysexec_checkcommand` return may be equal to: -1,0,1 where:
- 1 mean that command matched to one of predefined
- 0 mean that command is not permited
- (-1) mean that no commands were predefined thus all cammands are permited (no wldcrds file and no macro defined)

thats why acc is set to 0 in mentioned position - to propagate information that predefinitions exist and no match means failure. The function `psh_sysexec_checkcommand` as a whole can output only 0 or 1. Thats why `return (acc != 0);` is used.

#### About `argmaxlen`:
Intention was to secure the case of defining a wildcard command with too long argument. The longest word to be compared can be only as big as the defined buffer. But it is indeed over complicated and now i see that the `strlen(argv[ii]) != len` prevents too long argument (in fact any wrong-length argument) to be passed. So it is valid that this argument is now useless.

Reply:
My original comment was regarding why You check the `acc` value in `if` when You have set it to `0` in the previous line

Reply:
Check has been removed